flink-java/src/main/java/org/apache/flink/api/java/operators/PartitionOperator.java

```diff
@@ -98,6 +101,14 @@ public PartitionOperator(DataSet<T> input, Keys<T> pKeys, Partitioner<?> customP
 		this.customPartitioner = customPartitioner;
 		this.distribution = distribution;
 	}
+
+	public PartitionOperator<T> withOrders(Order... orders) {
```

Comment:
We should check that the number of `orders` is the same as the number of specified keys. Unfortunately, this is not as trivial as it sounds, because `Keys` does not give access to the the originally specified keys but only to the flattened logical keys. If a program specifies a `Tuple2<Long, Long>` as key, it will only specify a single order but the logical keys will be flattened to `[Long, Long]`. I think we should extend the `Keys` class by a method `TypeInformation<?>[] getOriginalKeyFieldTypes()` which returns the unflattened field types. Using that method we can see how many flat fields exist for each specified key field.